modules/capability/capability_test.go

```diff
@@ -3,101 +3,119 @@ package capability_test
 import (
 	"testing"
 
-	abci "github.com/cometbft/cometbft/abci/types"
-	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	"github.com/stretchr/testify/suite"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store"
+	"cosmossdk.io/store/metrics"
+	storetypes "cosmossdk.io/store/types"
+
+	dbm "github.com/cosmos/cosmos-db"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/types/module"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
-	"github.com/stretchr/testify/suite"
 
-	"github.com/cosmos/ibc-go/v7/testing/simapp"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
 
 	"github.com/cosmos/ibc-go/modules/capability"
 	"github.com/cosmos/ibc-go/modules/capability/keeper"
 	"github.com/cosmos/ibc-go/modules/capability/types"
 )
 
-const memStoreKey = "memory:mock"
+const mockMemStoreKey = "memory:mock"
 
 type CapabilityTestSuite struct {
 	suite.Suite
-	cdc    codec.Codec
-	ctx    sdk.Context
-	app    *simapp.SimApp
+
+	cdc codec.Codec
+	ctx sdk.Context
+
 	keeper *keeper.Keeper
-	module module.AppModule
+
+	storeKey        *storetypes.KVStoreKey
+	memStoreKey     *storetypes.MemoryStoreKey
+	mockMemStoreKey *storetypes.MemoryStoreKey
 }
 
 func (suite *CapabilityTestSuite) SetupTest() {
-	checkTx := false
-	app := simapp.Setup(suite.T(), checkTx)
-	cdc := app.AppCodec()
-
-	// create new keeper so we can define custom scoping before init and seal
-	keeper := keeper.NewKeeper(cdc, app.GetKey(types.StoreKey), app.GetMemKey(types.MemStoreKey))
-
-	suite.app = app
-	suite.ctx = app.BaseApp.NewContext(checkTx, tmproto.Header{Height: 1})
-	suite.keeper = keeper
-	suite.cdc = cdc
-	suite.module = capability.NewAppModule(cdc, *keeper, false)
+	encodingCfg := moduletestutil.MakeTestEncodingConfig(capability.AppModuleBasic{})
+	suite.cdc = encodingCfg.Codec
+
+	suite.storeKey = storetypes.NewKVStoreKey(types.StoreKey)
+	suite.memStoreKey = storetypes.NewMemoryStoreKey(types.MemStoreKey)
+	suite.mockMemStoreKey = storetypes.NewMemoryStoreKey(mockMemStoreKey)
+
+	suite.ctx = suite.NewTestContext()
+	suite.keeper = keeper.NewKeeper(suite.cdc, suite.storeKey, suite.memStoreKey)
+}
+
+func (suite *CapabilityTestSuite) NewTestContext() sdk.Context {
+	db := dbm.NewMemDB()
+	cms := store.NewCommitMultiStore(db, log.NewNopLogger(), metrics.NewNoOpMetrics())
+	cms.MountStoreWithDB(suite.storeKey, storetypes.StoreTypeIAVL, db)
+	cms.MountStoreWithDB(suite.memStoreKey, storetypes.StoreTypeMemory, db)
+	cms.MountStoreWithDB(suite.mockMemStoreKey, storetypes.StoreTypeMemory, db)
+
+	err := cms.LoadLatestVersion()
+	suite.Require().NoError(err)
+
+	return sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger())
 }
```

Comment:
This has been adapted from [here](https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.1/testutil/context.go).

I'm unsure why the test helpers in the sdk do not support adding mem store keys, or even mounting multiple keys (I guess nobody has run into an issue with the current code yet). 

I was going to make a PR to the sdk to add some extra support for this, but I felt it was cleaner and quicker to adapt it and maintain it here.

Reply:
Opened https://github.com/cosmos/cosmos-sdk/issues/16998